osquery/tables/system/windows/prefetch.cpp

```diff
@@ -0,0 +1,323 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include <osquery/core/core.h>
+#include <osquery/core/tables.h>
+#include <osquery/filesystem/filesystem.h>
+#include <osquery/logger/logger.h>
+#include <osquery/utils/conversions/join.h>
+#include <osquery/utils/conversions/tryto.h>
+#include <osquery/utils/conversions/windows/strings.h>
+#include <osquery/utils/conversions/windows/windows_time.h>
+#include <osquery/utils/windows/lzxpress.h>
+
+#include <boost/algorithm/hex.hpp>
+#include <boost/filesystem.hpp>
+
+namespace osquery {
+namespace tables {
+
+const std::string kPrefetchLocation = "C:\\Windows\\Prefetch\\";
+
+struct PrefetchHeader {
```

Comment:
i moved the utility functions to `prefetch.h` but tests will still be tricky to implement since prefetch is disabled on servers?

Reply:
I think we can still test things even if prefetch is not enabled on the server; before this can be merged, we should have some unit tests against some known well formed files and then some malformed ones.

The parsing functions should not have too much state so that once the PR is ready they can be easily hooked to oss-fuzz

Reply:
I read ur comment about adding prefetch files to `tools/tests` which is a good idea. Are there example tables that are doing something similar that i can reference? I see some test data in the directory (ex: plist files) but im not finding any tables that are utilizing the test files?
Would the test files/tables go somewhere else?